aether-ui/aether/ui/api/utils.py

```diff
@@ -171,212 +169,147 @@ def kernel_data_request(url='', method='get', data=None):
-        raise Exception(error)
-    if not pipeline.kernel_refs:
-        pipeline.kernel_refs = {}
-    if object_name is 'schema' or object_name is 'projectSchema':
-        if object_name not in pipeline.kernel_refs:
-            pipeline.kernel_refs[object_name] = {}
-        pipeline.kernel_refs[object_name][entity_name if entity_name is not None else data['name']] = res['id']
-        pipeline.save()
-        if object_name is 'schema':
-            create_project_schema_object('{}-{}'.format(project_name, data['name']),
-                                         pipeline, res['id'], data['name'])
-    else:
-        pipeline.kernel_refs[object_name] = res['id']
-        pipeline.save()
-    return pipeline
-
-
-def update_kernel_object(object_name, id, data):
-    return kernel_data_request(f'{object_name.lower()}s/{id}/', 'put', data)
-
-
 def is_object_linked(kernel_refs, object_name, entity_type_name=''):
     if kernel_refs and object_name in kernel_refs:
         try:
-            if object_name is 'schema' or object_name is 'projectSchema':
+            if object_name is 'schemas':
                 if entity_type_name in kernel_refs[object_name]:
-                    url = f'{object_name.lower()}s/{kernel_refs[object_name][entity_type_name]}'
+                    url = f'{object_name.lower()}/{kernel_refs[object_name][entity_type_name]}'
                 else:
                     return False
             else:
-                url = f'{object_name.lower()}s/{kernel_refs[object_name]}/'
-            kernel_data_request(url, 'get')
+                url = f'{object_name.lower()}/{kernel_refs[object_name]}/'
+            kernel_data_request(url)
             return True
         except Exception:
             return False
     else:
         return False
 
 
-def create_project_schema_object(name, pipeline, schema_id, entity_name):
-    if not is_object_linked(pipeline.kernel_refs, 'projectSchema', entity_name):
-        project_schema_data = {
-                            'name': name,
-                            'mandatory_fields': '[]',
-                            'transport_rule': '[]',
-                            'masked_fields': '[]',
-                            'is_encrypted': False,
-                            'project': pipeline.kernel_refs['project'],
-                            'schema': schema_id
-                        }
-        create_new_kernel_object('projectSchema',
-                                 pipeline, project_schema_data, entity_name=entity_name)
-    return pipeline
-
-
-def publish_preflight(pipeline, project_name, outcome):
+def publish_preflight(pipeline, outcome, contract):
     '''
     Performs a check for possible pipeline publish errors against kernel
     '''
-    if pipeline.mapping_errors:
-        outcome['error'].append('Mappings have errors')
+    if contract.mapping_errors:
+        outcome['error'].append('{} mappings have errors'.format(contract.name))
         return outcome
```

Comment:
Even with mapping errors, maybe it's a good option to continue checking the rest of artefacts and return all the problems together :thinking: 
_just thinking_

Reply:
Mapping errors are calculated during the pipeline edit, and are blockers for any publish. other checks require calls are made to the kernel and after a publish is initiated. i don't think the process should continue if there are mapping errors.

Reply:
My point is that the `input` only represents a possible scenario but not all of them and the mapping rules reflect one of them but not this one.

AVRO Schema
```
{
    "name": "Person",
    "type": "record",
    "fields": [
        {
            "name": "first_name",
            "type": "string",
        },
        {
            "name": "middle_name",
            "type": ["null", "string"],
        },
        {
            "name": "last_name",
            "type": "string",
        }
    ]
}
```
Input sample
```
{
  "name": "peter",
  "last_name": "smith"
}
```
Mapping rules:
```
['$.first_name', 'Person.first_name'],
['$.middle_name', 'Person.middle_name'],
['$.last_name', 'Person.last_name']
```

Reply:
But hey, it's just a comment. No action requested here.